index.js

```diff
@@ -91,16 +123,20 @@ module.exports = function renderDom(rootPostOrString, options, cb) {
 			resetPartial: resetPartial
 		}
 
-		butler.on('post changed', function (filename, post) {
+		butler.on('post changed', function(filename, post) {
 			if (partialExists(filename)) { // Only cares about posts that are in the system
 				if (filename === currentFilename) { // Current post changed
 					setCurrent(post)
 				} else {
 					scan(post, util, state, true)
 				}
 			}
+
+			posts[filename] = post
```

Comment:
yeah, it makes sense to move this assignment to the top of the function.  I don't know if I feel confident enough to remove line 92, though.